### PR TITLE
Simplify code of Worker-side RPC handling

### PR DESCRIPTION
**Pull request by apc999**
- Remove unnecessary class hierarchy: e.g., `DataServer`, `AbstractReadHandler`,  `ReadRequest`, `ReadRequestContext`
- Simplify calls `AsyncCacheRequestManager`

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/DataServer.java [hunk 1]
```diff
@@ -11,46 +11,15 @@
...
-     *
-     * @param hostName the host name of the data server
-     * @param dataAddress the address of the data server
-     * @param worker the Alluxio worker handle
-     * @return the generated {@link DataServer}
-     */
-    public static DataServer create(final String hostName,
-        final SocketAddress dataAddress, final WorkerProcess worker) {
-      return CommonUtils.createNewClassInstance(
-          ServerConfiguration.<DataServer>getClass(PropertyKey.WORKER_DATA_SERVER_CLASS),
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/DataServer.java:48]: there is only one implementation. I don't see why we want to classload DataServer

File: core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java [hunk 2]
```diff
@@ -99,16 +98,8 @@ public void submitRequest(AsyncCacheRequest request) {
...
-          long sessionId = isSourceLocal ? Sessions.ASYNC_CACHE_UFS_SESSION_ID
-              : Sessions.ASYNC_CACHE_WORKER_SESSION_ID;
           // Check if the block has already been cached on this worker
-          long lockId = mBlockWorker.lockBlock(sessionId, blockId);
-          if (lockId != BlockWorker.INVALID_LOCK_ID) {
-            try {
-              mBlockWorker.unlockBlock(lockId);
-            } catch (BlockDoesNotExistException e) {
-              LOG.error("Failed to unlock block on async caching. We should never reach here", e);
-            }
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java:111]: can be replaced by `hasBlockMeta`

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockWorkerClientServiceHandler.java [hunk 3]
```diff
@@ -58,8 +58,8 @@
  * Server side implementation of the gRPC BlockWorker interface.
  */
 @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
-public class BlockWorkerImpl extends BlockWorkerGrpc.BlockWorkerImplBase {
-  private static final Logger LOG = LoggerFactory.getLogger(BlockWorkerImpl.class);
+public class BlockWorkerClientServiceHandler extends BlockWorkerGrpc.BlockWorkerImplBase {
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockWorkerClientServiceHandler.java:61]: Follow master-side class name convention

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java [hunk 4]
```diff
@@ -19,72 +19,469 @@
...
+ *    there is, handle them properly. See more information about the notifications in the javadoc
+ *    of {@link ReadRequestContext} about CANCEL, EOF, and ERROR flags.
+ *
+ * @see ReadRequestContext
  */
-@edu.umd.cs.findbugs.annotations.SuppressFBWarnings(value = "BC_UNCONFIRMED_CAST_OF_RETURN_VALUE",
-    justification = "false positive with superclass generics, "
-        + "see more description in https://sourceforge.net/p/findbugs/bugs/1242/")
 @NotThreadSafe
-public final class BlockReadHandler extends AbstractReadHandler<BlockReadRequestContext> {
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java:58]: Combine `BlockReadHandler` and `AbstractReadHandler`. There is no other class extending `AbstractReadHandler`

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadRequest.java [hunk 5]
```diff
@@ -21,8 +22,13 @@
  * The internal representation of a block read request.
  */
 @ThreadSafe
-public final class BlockReadRequest extends ReadRequest {
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadRequest.java:24]: Combined with `ReadRequest.java` which has only one subclass now

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadRequestContext.java [hunk 6]
```diff
@@ -13,21 +13,224 @@
...
+
 import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
 import javax.annotation.concurrent.NotThreadSafe;
 
 /**
  * Context of {@link BlockReadRequest}.
  */
 @NotThreadSafe
-public final class BlockReadRequestContext extends ReadRequestContext<BlockReadRequest> {
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadRequestContext.java:23]: Combined with `ReadRequestContext.java` which has only one subclass now

File: core/server/worker/src/main/java/alluxio/worker/DataServer.java (hunk 1 above)
Comment (bf8086) [reply core/server/worker/src/main/java/alluxio/worker/DataServer.java:48]: IIRC there were more implementation pre-gRPC era. Now there is only `GrpcDataServer`.

**Review by bf8086 — APPROVED** [review]
LGTM. Thanks for making the code cleaner!

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**